### PR TITLE
ttop: 1.2.8 -> 1.5.2

### LINKED ISSUE #347429: Build failure: ttop
### Steps To Reproduce

1. Add ttop package to `environment.systemPackages`
2. Run `nixos-rebuild boot --flake`

### Build log
<details>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/56qq1gcjahdgzba324dp1lzf6qmv0d3h-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
generating config.nims
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build job 0: nim compile --path:"/nix/store/q3m2aqlzzrx4jj5akbf8rah0gp40ya2v-source/src" --path:"/nix/store/3lmm3z36qn4gz7bfa209zv0pqrpm3di9-source/" --path:"/nix/store/ila4vdklhqs6h14gwyx71yrjbzwf54g3-source/src" --path:"/nix/store/nrgl>
Hint: used config file '/nix/store/jnxnn4cmy4cw8pwi9qfxxmyyx1drc31p-x86_64-unknown-linux-gnu-nim-wrapper-2.2.0/etc/nim/nim.cfg' [Conf]
Hint: used config file '/nix/store/jnxnn4cmy4cw8pwi9qfxxmyyx1drc31p-x86_64-unknown-linux-gnu-nim-wrapper-2.2.0/etc/nim/config.nims' [Conf]
Hint: used config file '/build/source/config.nims' [Conf]
......................................................................................................................................
/build/source/src/ttop/config.nim(42, 16) Hint: simpleGetOrDefault(toml, "light") --> 'getOrDefault(toml, "light")' [Pattern]
/build/source/src/ttop/config.nim(44, 17) Hint: simpleGetOrDefault(toml, "docker") --> 'getOrDefault(toml, "docker")' [Pattern]
/build/source/src/ttop/config.nim(47, 16) Hint: simpleGetOrDefault(toml, "trigger") --> 'getOrDefault(toml, "trigger")' [Pattern]
/build/source/src/ttop/config.nim(48, 19) Hint: simpleGetOrDefault(t, "on_info") --> 'getOrDefault(t, "on_info")' [Pattern]
/build/source/src/ttop/config.nim(49, 20) Hint: simpleGetOrDefault(t, "on_alert") --> 'getOrDefault(t, "on_alert")' [Pattern]
/build/source/src/ttop/config.nim(53, 15) Hint: simpleGetOrDefault(t, "debug") --> 'getOrDefault(t, "debug")' [Pattern]
/build/source/src/ttop/config.nim(54, 13) Hint: simpleGetOrDefault(t, "cmd") --> 'getOrDefault(t, "cmd")' [Pattern]
.............
/build/source/src/ttop/procfs.nim(174, 11) template/generic instantiation of `catchErr` from here
/build/source/src/ttop/procfs.nim(178, 9) Error: expression 'uint(float(hz) * f)' is of type 'uint' and has to be used (or discarded); start of expression here: /build/source/src/ttop/procfs.nim(174, 11)
build phase failed

```
</details>

### Additional context

For build package I'm using nixos-unstable branch.

<details>
<summary>flake.lock</summary>

```
...
    "nixpkgs": {
      "locked": {
        "lastModified": 1728241625,
        "narHash": "sha256-yumd4fBc/hi8a9QgA9IT8vlQuLZ2oqhkJXHPKxH/tRw=",
        "owner": "nixos",
        "repo": "nixpkgs",
        "rev": "c31898adf5a8ed202ce5bea9f347b1c6871f32d1",
        "type": "github"
      },
      "original": {
        "owner": "nixos",
        "ref": "nixos-unstable",
        "repo": "nixpkgs",
        "type": "github"
      }
    },
...

```

</details>

### Notify maintainers

@figsoda @sikmir

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.11-lqx1, NixOS, 24.11 (Vicuna), 24.11.20241006.c31898a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.0
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/ulad/.config/nix/nix.conf:/nix/store/94f63l1cwclzaddsscc89bsw9g1pjj4n-cinnamon-settings-daemon-6.2.0/etc/xdg/nix/nix.conf:/nix/store/94f63l1cwclzaddsscc89bsw9g1pjj4n-cinnamon-settings-daemon-6.2.0/etc/xdg/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/ulad/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/ulad/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/ulad/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/cfl6rqw64ndbh2m0qapg410jf4xmycia-lix-2.91.0/share`
 - nixpkgs: `/nix/store/rs4fjbnw4qx7ns2hzzrz2iz52va7vs5z-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by jackwilsdon**
Fixes #347429.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tt/ttop/package.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
   pname = "ttop";
-  version = "1.2.8";
+  version = "1.5.2";
 
   src = fetchFromGitHub {
     owner = "inv2004";
     repo = "ttop";
     rev = "v${finalAttrs.version}";
-    hash = "sha256-QMUrA3OjxlDa1OxptJL7T3SPDTzSwVz6zz+ueh9eovM=";
+    hash = "sha256-4jkVNfyEJ3CdCVe9z+bfj/63AU+HhNYgH4OEj1rl9Xo=";
```
Comment (sikmir) [inline pkgs/by-name/tt/ttop/package.nix:11]:
```
error: hash mismatch in fixed-output derivation '/nix/store/cr1c7ay8ybglr11fq2xf1kqk8i6vjilp-source.drv':
         specified: sha256-4jkVNfyEJ3CdCVe9z+bfj/63AU+HhNYgH4OEj1rl9Xo=
            got:    sha256-/rs5JjTXxptVHXL3fY8qP6Be3r5N871CEbUH7w6zx4A=
```

Comment (jackwilsdon) [reply pkgs/by-name/tt/ttop/package.nix]: It looks like upstream just re-tagged 1.5.2 :grimacing: https://github.com/inv2004/ttop/releases/tag/v1.5.2

It previously pointed at https://github.com/inv2004/ttop/commit/f0bc9c5663ca7aff679127d3340c77d0133ebe34 but was updated to https://github.com/inv2004/ttop/commit/214fcd123d03798c90b74f3b03b1556678fc7aab (comparison [here](https://github.com/inv2004/ttop/compare/f0bc9c5663ca7aff679127d3340c77d0133ebe34...214fcd123d03798c90b74f3b03b1556678fc7aab)).

I've updated the hash.

**Review by sikmir — APPROVED** [review]
Result of `nixpkgs-review pr 347577` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ttop</li>
  </ul>
</details>

**Comment by teto** [comment]
Result of `nixpkgs-review pr 347577` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ttop</li>
  </ul>
</details>

**Comment by teto** [comment]
not merging because I've seen some urls change in the lockfile and dunno the program enough to know if it's valid. But it runs at leastr.

**Review by fpletz — APPROVED** [review]

**Pull request merged**